mcs/class/corlib/System.Reflection/CustomAttributeTypedArgument.cs

```diff
@@ -82,17 +84,46 @@ public CustomAttributeTypedArgument (object value)
 			}
 		}
 
-		public override string ToString ()
+		public override string ToString () => ToString (typed: false);
+
+		internal string ToString (bool typed)
 		{
-			string val = value != null ? value.ToString () : String.Empty;
-			if (argumentType == typeof (string))
-				return "\"" + val + "\"";
-			if (argumentType == typeof (Type)) 
-				return "typeof (" + val + ")";
-			if (argumentType.IsEnum)
-				return "(" + argumentType.Name + ")" + val;
-
-			return val;
+			if (ArgumentType == null)
+				return base.ToString (); // Someone called ToString() on "default(CustomAttributeTypedArgument)"
+
+			try	{
+				if (ArgumentType.IsEnum)
+					return string.Format (CultureInfo.CurrentCulture, typed ? "{0}" : "({1}){0}", Value, ArgumentType.FullName);
+
+				else if (Value == null)
```

Comment:
else(s) are redundant

Reply:
Not sure why

Reply:
You have returns before

Reply:
ok, I thought you meant the all `else` branches